src/lib/helpers.ts

```diff
@@ -344,3 +355,338 @@ export function parseResponseAPIVersion(response: Response) {
+     */
+    transports: descriptor.transports as AuthenticatorTransport[],
+  }
+}
+
+/**
+ * If possible coerce a `string` value into a known `AuthenticatorAttachment`
+ */
+function toAuthenticatorAttachment(attachment: string | null): AuthenticatorAttachment | undefined {
+  const attachments: AuthenticatorAttachment[] = ['cross-platform', 'platform']
+
+  if (!attachment) {
+    return
+  }
+
+  if (attachments.indexOf(attachment as AuthenticatorAttachment) < 0) {
+    return
+  }
+
+  return attachment as AuthenticatorAttachment
+}
+
+/**
+ * Visibly warn when we detect an issue related to a passkey provider intercepting WebAuthn API
+ * calls
+ */
+function warnOnBrokenImplementation(methodName: string, cause: Error): void {
+  console.warn(
+    `The browser extension that intercepted this WebAuthn API call incorrectly implemented ${methodName}. You should report this error to them.\n`,
+    cause
+  )
+}
+
+class BaseWebAuthnAbortService {
+  private controller: AbortController | undefined
+
+  /**
+   * Prepare an abort signal that will help support multiple auth attempts without needing to
+   * reload the page. This is automatically called whenever `startRegistration()` and
+   * `startAuthentication()` are called.
+   */
+  createNewAbortSignal() {
+    // Abort any existing calls to navigator.credentials.create() or navigator.credentials.get()
+    if (this.controller) {
+      const abortError = new Error('Cancelling existing WebAuthn API call for new one')
+      abortError.name = 'AbortError'
+      this.controller.abort(abortError)
+    }
+
+    const newController = new AbortController()
+
+    this.controller = newController
+    return newController.signal
+  }
+
+  /**
+   * Manually cancel any active WebAuthn registration or authentication attempt.
+   */
+  cancelCeremony() {
+    if (this.controller) {
+      const abortError = new Error('Manually cancelling existing WebAuthn API call')
+      abortError.name = 'AbortError'
+      this.controller.abort(abortError)
+
+      this.controller = undefined
+    }
+  }
+}
+
+/**
+ * A service singleton to help ensure that only a single WebAuthn ceremony is active at a time.
+ *
+ * Users of **@simplewebauthn/browser** shouldn't typically need to use this, but it can help e.g.
+ * developers building projects that use client-side routing to better control the behavior of
+ * their UX in response to router navigation events.
+ */
+export const WebAuthnAbortService = new BaseWebAuthnAbortService()
+
+/**
+ * Begin authenticator "registration" via WebAuthn attestation
+ *
+ * @param optionsJSON Output from **@simplewebauthn/server**'s `generateRegistrationOptions()`
+ */
+export async function startRegistration(
+  optionsJSON: PublicKeyCredentialCreationOptionsJSON
+): Promise<RegistrationResponseJSON> {
+  if (!browserSupportsWebAuthn()) {
+    throw new Error('WebAuthn is not supported in this browser')
```

Comment:
Add additional context here... something like:

```suggestion
    throw new Error('WebAuthn is not supported in this browser. If using in SSR please use this in client components or pages.')
```